### ISSUE TITLE
Add `crypto/x509` to this repo

### ISSUE THREAD

**Issue by cuthix** [ProtonMail/crypto#16]
Right now we are not able to test `go test ./openpgp/packet/`

```
# golang.org/x/crypto/openpgp/packet [golang.org/x/crypto/openpgp/packet.test]
openpgp/packet/private_key_test.go:132:37: cannot use rsaPriv (type *"crypto/rsa".PrivateKey) as type *"golang.org/x/crypto/rsa".PrivateKey in argument to NewRSAPrivateKey
openpgp/packet/private_key_test.go:224:33: not enough arguments in call to "golang.org/x/crypto/rsa".GenerateKey
        have (io.Reader, number)
        want (io.Reader, int, []*big.Int)
FAIL    golang.org/x/crypto/openpgp/packet [build failed]
```

even though this two classes have same structure  go treat them as two uniq types.

**Comment by cuthix** [comment]
@sanjanarajan @twiss

Current state of https://github.com/golang/crypto (`github.com/golang/x/crypto`) and our repo is very different. The major difference is that stable parts has been moved to official go release https://golang.org/pkg/crypto/#pkg-subdirectories (`GOROOT/crypto`) (e.g. rsa)

We have two options:
1. Keep the current structure and update subdirectories from `GOROOT/crypto`
2. Merge to current `github.com/golang/x/crypto`

To sum it up
1. is probably less work now, but more work later (we need to keep track changes from different sources and if we want to be merge back to original we have to do 2. anyway)
2. is more work now (it might impact also `go-pm-crypto`) but less work later

**Comment by cuthix** [comment]
I personally prefer 2.

**Comment by twiss** [comment]
No longer needed since https://github.com/ProtonMail/crypto/pull/19/commits/7abb2214c394172848607b2ad06082dc9715e051#diff-1968450dba45ce9296aad32d115bb1d9 - x509 was only used in tests.
